packages/x-charts-pro/src/context/ZoomProvider/useSetupPan.ts

```diff
@@ -0,0 +1,115 @@
+import * as React from 'react';
+import { useDrawingArea, useSvgRef } from '@mui/x-charts/hooks';
+import { getSVGPoint } from '@mui/x-charts/internals';
+import { useZoom } from './useZoom';
+
+const MAX_RANGE = 100;
+const MIN_RANGE = 0;
+
+// TODO: move to helper
+const isPointOutside = (
+  point: { x: number; y: number },
+  area: { left: number; top: number; width: number; height: number },
+) => {
+  const outsideX = point.x < area.left || point.x > area.left + area.width;
+  const outsideY = point.y < area.top || point.y > area.top + area.height;
+  return outsideX || outsideY;
+};
+
+export const useSetupPan = () => {
+  const { zoomRange, setZoomRange, setIsInteracting } = useZoom();
+  const area = useDrawingArea();
+
+  const svgRef = useSvgRef();
+
+  const isDraggingRef = React.useRef(false);
+  const touchStartRef = React.useRef<{ x: number; minX: number; maxX: number } | null>(null);
+  const eventCacheRef = React.useRef<PointerEvent[]>([]);
+
+  React.useEffect(() => {
+    const element = svgRef.current;
+    if (element === null) {
+      return () => {};
+    }
+
+    const handlePan = (event: PointerEvent) => {
+      if (element === null || !isDraggingRef.current || eventCacheRef.current.length > 1) {
+        return;
+      }
+
+      if (touchStartRef.current == null) {
+        return;
+      }
+
+      const point = getSVGPoint(element, event);
+      const movementX = point.x - touchStartRef.current.x;
+
+      if (isPointOutside(point, area)) {
+        isDraggingRef.current = false;
+        setIsInteracting(false);
+        return;
+      }
```

Comment:
Yeah trying to fix it up now. Currently the behaviours block the mobile tooltip. I'll move mobile tooltips to pointerevents, since other ways of blocking the default pinch behaviour (zoom page) either require changing the head, or disables all touch actions...